### DISCUSSION TITLE
AM550 config for Slovenia (encrypted)

### DISCUSSION THREAD

**Discussion by davidg-0** [UtilitechAS/amsreader-firmware#858] [Q&A] [answered]
Hi, 
I'm using WT32-ETH01 via Ethernet, specifically this board (https://github.com/zuidwijk/slimmelezer-wt32-eth01)
Meter: Iskraemeco AM550 with P1 port (AM550-TD1.21)
Country: Slovenia
Communication is encrypted+authenticated.

The board powers up via P1 port, but there are no readings showing.
I'm not sure if I've put wrong settings, or is this combination of meter/country not supported by AMS yet.

Could you please help me troubleshoot this setup?

Slovenia is currently rolling out dynamic - block based power pricing, so I'm sure I'm not the only one trying make this setup work.

Appreciate your work!

![AMS-AM550](https://github.com/user-attachments/assets/99df718f-f1a6-4efb-af54-79dc370fa1b7)

**Comment by ArnieO** [comment]
Try a Telnet debug to see if there is any data coming from the meter: https://www.amsleser.no/module/ets_blog/blog?id_post=24

**Reply by davidg-0** [reply]
I can't find any "System/Configuration" menu on the latest firmware (v2.3.8), as you can see on the screenshot. 
Should I use some other one?

**Reply by ArnieO** [reply]
Sorry!
I read too fast yesterday, and did not notice you are on an Ethernet solution. The page you show is indeed the configuration screen, but the Ethernet version does not have the Telnet debug option. 

The Ethernet option is currently an "addon for DIYers" to our firmware. 
I have no hands-on experience with it, so I cannot guide you.
Maybe @gskjold could guide you on how to access the raw data (if any) from the meter.

**Reply by davidg-0** [reply]
No problem, I was not aware of that. I just flashed the latest version from releases.
I tried a Factory Reset and made a WiFi connection, but there's still no menu available.

**Reply by ArnieO** [reply]
Ah... I can see I have to update the article! The first screenshot is from a quite old firmware version. 
Go to the config page and look at the bottom. You should find the "Debugging" tile there:
![image](https://github.com/user-attachments/assets/a2a45e7a-db67-4a29-85de-15c6567d6988)

**EDIT:**
The article is updated now, thank you!

**Reply by davidg-0** [reply]
I'm sorry. There's still no "Debugging" tile.
I do see a new "HAN: Parse error", when using WiFi though. Does that help to indicate anything?
![AMS-AM550-WiFi](https://github.com/user-attachments/assets/0174e895-408b-4367-be66-d736c2e8c091)

**Reply by ArnieO** [reply]
Ping @gskjold

**Reply by zitko** [reply]
Joining this thread as I'm also from Slovenia (I assume I have the same meter model as @davidg-0) and also have problems reading my Iskraemeco AM550 energy meter.

I recently received POW-P1 (great device, btw!) but only today managed to get my energy distributor to enable unencrypted P1 port on the energy meter. I have POW-P1 configured at 38400/8N1.

It can read incoming frames but seemingly not all of them (no errors in the logs though).

The dashboard only shows meter type (Iskra) and (summary?) consumption (next to the Iskra label). HAN indicator is colored red. There are no parse (or other) errors shown.

What it doesn't show is current/past input and/or power consumption.

Based on the debugs it's receiving HDLC/LLC/DLMS frames every second.

This is an example DLMS frame:

```
<DataNotification>
  <LongInvokeIdAndPriority Value="000BF74B" />
  <!--2024-11-19 16:58:18-->
  <DateTime Value="07E80B1302103A1200FFC400" />
  <NotificationBody>
    <DataValue>
      <Structure Qty="0A" >
        <!--ISK1020785766649-->
        <OctetString Value="49534B31303230373835373636363439" />
        <!--ISK85766649-->
        <OctetString Value="49534B3835373636363439" />
        <UInt32 Value="000002EC" />
        <UInt32 Value="00000000" />
        <UInt32 Value="00000000" />
        <UInt32 Value="00000164" />
        <UInt16 Value="093E" />
        <UInt16 Value="0165" />
        <UInt32 Value="000002EC" />
        <UInt32 Value="00000000" />
      </Structure>
    </DataValue>
  </NotificationBody>
</DataNotification>
```

after this frame I get this in the logs:

```
(D) Received valid DLMS at 30
(V) Using application data:
(V) 02 0A 09 10  49 53 4B 31  30 32 30 37  38 35 37 36
(V) 36 36 34 39  09 0B 49 53  4B 38 35 37  36 36 36 34
(V) 39 06 00 00  02 EC 06 00  00 00 00 06  00 00 00 00
(V) 06 00 00 01  64 12 09 3E  12 01 65 06  00 00 02 EC
(V) 06 00 00 00  00
(V) DLMS
```

One more thing, the following warning is dumped to the logs every 30 seconds:

`(W) No HAN data received last 30s, single blink`

Not sure if any of the above helps.

I too would really like to see this working.

If I can be of any additional help (debugs & such) please let me know.

**Reply by zitko** [reply]
just realized there's a discussion on reading Slovenian power meters in ticket [https://github.com/UtilitechAS/amsreader-firmware/issues/869](https://github.com/UtilitechAS/amsreader-firmware/issues/869), will follow there

**Reply by ArnieO** [reply]
Yes, please use that other thread. Slovenia is difficult - for the reasons explained there. 
We will need information similar to what has been provided from some users in https://github.com/UtilitechAS/amsreader-firmware/issues/615 to handle the different types of Slovenian payloads.

**Comment by davidg-0** [comment, marked as answer]
I think I managed to make it work.
My grid company is "Elektro Ljubljana".

I had wrong baud rate set, since I referenced Wiki/Known hardware configurations
I found the correct values for Slovenia (38400/8N1/512/non inverted) here:
https://github.com/UtilitechAS/amsreader-firmware/issues/869#issuecomment-2493771861

Thanks again for the open firmware, and kindness to help with non standard configs!
![AMS-info](https://github.com/user-attachments/assets/e9f27d09-ea6d-4e4f-9403-06dad0e921bb)

**Reply by ArnieO** [reply]
Thank you for the good news! Yes, Slovenia seems to use 38400 baud. 
I have added a separate line in the table now - for Slovenia. https://github.com/UtilitechAS/amsreader-firmware/wiki/Known-hardware-configurations
